domain/src/main/java/com/teamagam/gimelgimel/domain/messages/AddPolledMessageToRepositoryInteractor.java

```diff
@@ -55,27 +53,26 @@ public AddPolledMessageToRepositoryInteractor(
         return observable.doOnNext(mMessagesRepository::putMessage);
     }
 
-    private Observable<Boolean> updateUnreadCountRepository(Observable<Message> observable) {
-        return observable.flatMap(this::shouldHandleMessageAsUnread)
-                .filter(shouldHandleAsUnread -> shouldHandleAsUnread)
-                .doOnNext(shouldHandleAsUnread -> mUnreadMessagesCountRepository.addNewUnreadMessage());
+    private Observable<?> updateUnreadCountRepository(Observable<Message> observable) {
+        return observable
+                .doOnNext(message -> {
+                    if (shouldHandleMessageAsUnread(message)) {
+                        mUnreadMessagesCountRepository.addNewUnreadMessage(message.getCreatedAt());
+                    }
+                });
     }
 
-    private Observable<Boolean> shouldHandleMessageAsUnread(Message message) {
-        return mMessagesContainerStateRepository.getState()
-                .map(state -> !isVisible(state) && !isFromSelf(message) && !alreadyRead(message));
+    private boolean shouldHandleMessageAsUnread(Message message) {
+        return !isFromSelf(message) && !alreadyRead(message);
     }
 
     private boolean alreadyRead(Message message) {
-        return mUnreadMessagesCountRepository.getLastVisitTimestamp().after(message.getCreatedAt());
-    }
-
-    private boolean isVisible(MessagesContainerStateRepository.ContainerState state) {
-        return state == MessagesContainerStateRepository.ContainerState.VISIBLE;
+        Date systemDate = mUnreadMessagesCountRepository.getLastVisitTimestamp();
+        Date messageDate = message.getCreatedAt();
+        return systemDate.after(messageDate) || systemDate.equals(messageDate);
```

Comment:
Won't work....
before and after are exclusive
I can do !systemDate.before(messageDate), but it's somewhat unclear